src/payments/services/payment.service.ts

```
@@ -395,23 +395,89 @@ export class PaymentService {
     }
+  }
 
-    return result;
+  /**
+   * One certificate per distinct course (contextId), then a single unlock for the whole intent.
+   * Avoids duplicate external certificate calls and repeated unlockAll when targets share a context.
+   */
+  private async issueCertificateBundleForContexts(
+    paymentIntentId: string,
+    userId: string,
+    contextIds: string[],
```

ESLint

[error] Delete `,`

(prettier/prettier)